src/Raven.Server/Web/System/AdminDatabasesHandler.cs

```diff
@@ -242,6 +253,87 @@ public bool NotUsingHttps(string url)
             }
         }
 
+        private async Task RecreateIndexes(DatabaseRecord databaseRecord)
+        {
+            PathSetting dataDirectory = databaseRecord.Settings.TryGetValue("DataDir", out var path) == false || path == null
+                ? new PathSetting(ServerStore.Configuration.Core.DataDirectory.ToFullPath(), ResourceType.Database, databaseRecord.DatabaseName) : new PathSetting(path);
+
+            if (Directory.Exists(dataDirectory.FullPath) == false)
+            {
+                return;
+            }
+
+            var addToInitLog = new Action<string>(txt =>
+            {
+                var msg = $"[Recreateing indexes] {DateTime.UtcNow} :: Database '{databaseRecord.DatabaseName}' : {txt}";
+                if (Logger.IsInfoEnabled)
+                    Logger.Info(msg);
+            });
+
+            using (var documnetDatabase = new DocumentDatabase(databaseRecord.DatabaseName,
+                new RavenConfiguration(databaseRecord.DatabaseName, ResourceType.Database)
+                {
+                    Core =
+                    {
+                            DataDirectory = dataDirectory,
+                            RunInMemory = false
+                    }
+                }, ServerStore, addToInitLog))
+            {
+                var options = InitializeOptions.SkipLoadingDatabaseRecord;
+                documnetDatabase.Initialize(options);
+
+                var command = new GetRaftEtagCommand();
+
+                var indexesPath = dataDirectory.FullPath + "\\Indexes";
+                foreach (var indexPath in Directory.GetDirectories(indexesPath))
```

Comment:
should use configuration.Indexing.Path here, not do any combines on your own

Reply:
the other reason is that the indexes path can be changed server-wide, so it should be taken into account. This is why it is better to create a configuration via databasesLandlord

Reply:
fixed